storage/badger/epoch_commits_test.go

```diff
@@ -36,7 +45,135 @@ func TestEpochCommitStoreAndRetrieve(t *testing.T) {
+	rlpEncodable := struct {
+		Counter            uint64
+		ClusterQCs         []flow.ClusterQCVoteData
+		DKGGroupKey        []byte
+		DKGParticipantKeys [][]byte
+	}{
+		Counter:            commit.Counter,
+		ClusterQCs:         commit.ClusterQCs,
+		DKGGroupKey:        commit.DKGGroupKey.Encode(),
+		DKGParticipantKeys: make([][]byte, 0, len(commit.DKGParticipantKeys)),
+	}
+	for _, key := range commit.DKGParticipantKeys {
+		rlpEncodable.DKGParticipantKeys = append(rlpEncodable.DKGParticipantKeys, key.Encode())
+	}
+
+	return rlp.Encode(w, rlpEncodable)
+}
+
+// encodableCommit represents encoding of epochCommitV0, it is used for serialization purposes and is used only in tests.
+// TODO(EFM, #6794): Remove this once we complete the network upgrade
+type encodableCommit struct {
+	Counter            uint64
+	ClusterQCs         []flow.ClusterQCVoteData
+	DKGGroupKey        encodable.RandomBeaconPubKey
+	DKGParticipantKeys []encodable.RandomBeaconPubKey
+}
+
+func encodableFromCommit(commit *epochCommitV0) encodableCommit {
+	encKeys := make([]encodable.RandomBeaconPubKey, 0, len(commit.DKGParticipantKeys))
+	for _, key := range commit.DKGParticipantKeys {
+		encKeys = append(encKeys, encodable.RandomBeaconPubKey{PublicKey: key})
+	}
+	return encodableCommit{
+		Counter:            commit.Counter,
+		ClusterQCs:         commit.ClusterQCs,
+		DKGGroupKey:        encodable.RandomBeaconPubKey{PublicKey: commit.DKGGroupKey},
+		DKGParticipantKeys: encKeys,
+	}
+}
+
+func commitFromEncodable(enc encodableCommit) epochCommitV0 {
+	dkgKeys := make([]crypto.PublicKey, 0, len(enc.DKGParticipantKeys))
+	for _, key := range enc.DKGParticipantKeys {
+		dkgKeys = append(dkgKeys, key.PublicKey)
+	}
+	return epochCommitV0{
+		Counter:            enc.Counter,
+		ClusterQCs:         enc.ClusterQCs,
+		DKGGroupKey:        enc.DKGGroupKey.PublicKey,
+		DKGParticipantKeys: dkgKeys,
+	}
+}
+
+func (commit *epochCommitV0) MarshalMsgpack() ([]byte, error) {
+	return msgpack.Marshal(encodableFromCommit(commit))
+}
+
+func (commit *epochCommitV0) UnmarshalMsgpack(b []byte) error {
+	var enc encodableCommit
+	err := msgpack.Unmarshal(b, &enc)
+	if err != nil {
+		return err
+	}
+	*commit = commitFromEncodable(enc)
+	return nil
+}
+
+// TestStoreV0AndDecodeV1 tests that an [flow.EpochCommit] without [flow.DKGIndexMap](v0) field can be stored and
+// later retrieved as a [flow.EpochCommit](v1) without any errors or data loss.
+// This test verifies that the [flow.EpochCommit] is backward compatible with respect to the [flow.DKGIndexMap] field.
+// TODO(EFM, #6794): Remove this once we complete the network upgrade
+func TestStoreV0AndDecodeV1(t *testing.T) {
+	unittest.RunWithBadgerDB(t, func(db *badger.DB) {
+		v1 := unittest.EpochCommitFixture()
+		v0 := &epochCommitV0{
+			Counter:            v1.Counter,
+			ClusterQCs:         v1.ClusterQCs,
+			DKGGroupKey:        v1.DKGGroupKey,
+			DKGParticipantKeys: v1.DKGParticipantKeys,
+		}
+		require.Equal(t, v0.ID(), v1.ID())
```

Comment:
The problem is we can create only an non-empty `DKGIndexMap` but not a valid one since `EpochCommitFixture` doesn't know anything about participants unless we require that caller needs to be pass participants for each call.